segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -212,6 +219,25 @@ public void setupAppend(SetupAppend setupAppend) {
                         });
     }
 
+    private void populateEventSizeForIndexSegmentAppends(UUID writer, String indexSegment, long requestId) {
+        store.getStreamSegmentInfo(indexSegment, TIMEOUT)
+                .whenComplete((properties, u) -> {
+                    if (u != null) {
+                        u = Exceptions.unwrap(u);
```

Comment:
Can you please add test case for exception scenarios?

Reply:
will be raising it in a new PR